azkaban-common/src/main/java/azkaban/executor/container/ContainerCleanupManager.java

```diff
@@ -134,15 +136,45 @@ public void cleanUpStaleFlows() {
   }
 
   /**
-   * Try to clean the stale containers that are older than maximum azkaban flow running time
+   * From all pods in the current namespace and Azkaban cluster, select the pods whose executions
+   * are not in unfinished statuses, e.g. DISPATCHING, PREPARING, RUNNING, PAUSED, KILLING,
+   * FAILED_FINISHING in DB, and delete the pods & services to release resources
    */
-  public void cleanUpStaleContainers() {
-    Duration containerValidity = validityMap.get(Status.RUNNING).getFirst();
-    if (!containerValidity.isNegative()) {
+  public void cleanUpContainersInTerminalStatuses() {
+    logger.info("Cleaning up pods of terminated flow executions");
+    Set<Integer> activeFlows = new HashSet<>();
+    Set<Integer> currentNameSpacedPods = new HashSet<>();
+    ImmutableMap<Status, Pair<Duration, String>> unFinishedStatusesMap =
```

Comment:
Should READY and QUEUED be included?
I think we can reuse and specify the public static unfinished set in Status.java similar to what we did in "nonFinishingStatusAfterFlowStartsSet" there.

Reply:
When flow is in READY or QUEUED status, the pods have not been created. Since we're only interested in cleaning up leaking pods, querying db for these two statuses is not needed.